pkg/resources/task.go

```diff
@@ -0,0 +1,645 @@
+		err := snowflake.Exec(db, q)
+		if err != nil {
+			return errors.Wrapf(err, "error updating comment on task %v", data.Id())
+		}
+		data.SetPartial("comment")
+	}
+
+	if data.HasChange("after") {
+		var (
+			q        string
+			err      error
+			old, new = data.GetChange("after")
+			enabled  = data.Get("enabled").(bool)
+		)
+
+		if enabled {
+			q = builder.Suspend()
+			err = snowflake.Exec(db, q)
+			if err != nil {
+				return errors.Wrapf(err, "error suspending task %v", data.Id())
+			}
+			defer resumeTask(builder, meta)
+		}
+
+		if old != "" {
+			q = builder.RemoveDependency(old.(string))
+			err = snowflake.Exec(db, q)
+			if err != nil {
+				return errors.Wrapf(err, "error removing old after dependency from task %v", data.Id())
+			}
+		}
+
+		if new != "" {
+			q = builder.AddDependency(new.(string))
+			err := snowflake.Exec(db, q)
+			if err != nil {
+				return errors.Wrapf(err, "error adding after dependency on task %v", data.Id())
+			}
+		}
+
+		data.SetPartial("after")
+	}
+
+	if data.HasChange("session_parameters") {
+		var q string
+		o, n := data.GetChange("session_parameters")
+
+		if o == nil {
+			o = make(map[string]interface{})
+		}
+		if n == nil {
+			n = make(map[string]interface{})
+		}
+		os := o.(map[string]interface{})
+		ns := n.(map[string]interface{})
+
+		remove := difference(os, ns)
+		add := difference(ns, os)
+
+		if len(remove) > 0 {
+			q = builder.RemoveSessionParameters(remove)
+			err := snowflake.Exec(db, q)
+			if err != nil {
+				return errors.Wrapf(err, "error removing session_parameters on task %v", data.Id())
+			}
+		}
+
+		if len(add) > 0 {
+			q = builder.AddSessionParameters(add)
+			err := snowflake.Exec(db, q)
+			if err != nil {
+				return errors.Wrapf(err, "error adding session_parameters to task %v", data.Id())
+			}
+		}
+
+		data.SetPartial("session_parameters")
+	}
+
+	if data.HasChange("when") {
+		_, new := data.GetChange("when")
+		q := builder.ChangeCondition(new.(string))
+		err := snowflake.Exec(db, q)
+		if err != nil {
+			return errors.Wrapf(err, "error updating when condition on task %v", data.Id())
+		}
+		data.SetPartial("when")
+	}
+
+	if data.HasChange("sql_statement") {
+		_, new := data.GetChange("sql_statement")
+		q := builder.ChangeSqlStatement(new.(string))
+		err := snowflake.Exec(db, q)
+		if err != nil {
+			return errors.Wrapf(err, "error updating sql statement on task %v", data.Id())
+		}
+		data.SetPartial("sql_statement")
+	}
+
+	if data.HasChange("enabled") {
+		var (
+			q      string
+			_, n   = data.GetChange("enabled")
+			enable = n.(bool)
+		)
+
+		if enable {
+			q = builder.Resume()
+		} else {
+			q = builder.Suspend()
+		}
+
+		err := snowflake.Exec(db, q)
+		if err != nil {
+			return errors.Wrapf(err, "error updating task state %v", data.Id())
+		}
+		data.SetPartial("enabled")
+	}
+
+	return ReadTask(data, meta)
+}
+
+// DeleteTask implements schema.DeleteFunc
+func DeleteTask(data *schema.ResourceData, meta interface{}) error {
```

Comment:
@henriblancke you need to handle root task suspend/resume here as well

Reply:
@chriskuchin it should be handled?

Reply:
Yeah you need to suspend the root to delete a child task

Reply:
@chriskuchin thats what should be happening at line 600 here

Reply:
yeah you are good I just missed those lines